crates/sui-graphql-rpc/src/types/transaction_block/tx_lookups.rs

```diff
@@ -102,65 +113,85 @@ impl TxBounds {
             None => None,
         };
 
-        // Upperbound in terms of checkpoint sequence number. We want to get the total transaction
-        // count at the end of this checkpoint. If no upperbound is given, use
-        // `checkpoint_viewed_at`.
+        // Inclusive upper bound in terms of checkpoint sequence number. If no upperbound is given,
+        // use `checkpoint_viewed_at`.
         //
         // SAFETY: we can unwrap because of the `Some(checkpoint_viewed_at)
         let cp_hi = min_option([cp_before_inclusive, cp_at, Some(checkpoint_viewed_at)]).unwrap();
 
+        // Read from the `checkpoints` table rather than the `pruner_cp_watermark` table, because
```

Comment:
Is there an instance at which we read from `pruner_cp_watermark`? (In other words, can we get rid of that table?).

Reply:
yes it's used by the indexer, although just proposed a change in data platform channel that could allow us to get rid of this outright

Reply:
agreed that these are duplicate info between pruner_cp_watermark and checkpoints, and I was looking at the de-dup, but prob no longer relevant given indexer-alt.